homeassistant/components/xbox/media_player.py

```diff
@@ -0,0 +1,241 @@
+)
+from homeassistant.const import (
+    STATE_OFF,
+    STATE_ON,
+    STATE_PAUSED,
+    STATE_PLAYING,
+    STATE_UNKNOWN,
+)
+
+from .const import APP_LEGACY_MAP, DOMAIN, HOME_LEGACY_PRODUCT_ID
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_XBOX = (
+    SUPPORT_TURN_ON
+    | SUPPORT_TURN_OFF
+    | SUPPORT_PREVIOUS_TRACK
+    | SUPPORT_NEXT_TRACK
+    | SUPPORT_PLAY
+    | SUPPORT_PAUSE
+    | SUPPORT_VOLUME_STEP
+    | SUPPORT_VOLUME_MUTE
+)
+
+XBOX_STATE_MAP = {
+    PlaybackState.Playing: STATE_PLAYING,
+    PlaybackState.Paused: STATE_PAUSED,
+    PowerState.On: STATE_ON,
+    PowerState.SystemUpdate: STATE_OFF,
+    PowerState.ConnectedStandby: STATE_OFF,
+    PowerState.Off: STATE_OFF,
+    PowerState.Unknown: STATE_UNKNOWN,
+}
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Xbox media_player from a config entry."""
+    client: XboxLiveClient = hass.data[DOMAIN][entry.entry_id]
+    consoles: SmartglassConsoleList = await client.smartglass.get_console_list()
+    async_add_entities(
+        [XboxMediaPlayer(client, console) for console in consoles.result], True
+    )
+
+
+class XboxMediaPlayer(MediaPlayerEntity):
+    """Representation of an Xbox device."""
+
+    def __init__(self, client: XboxLiveClient, console: SmartglassConsole) -> None:
+        """Initialize the Plex device."""
+        self.client: XboxLiveClient = client
+        self._console: SmartglassConsole = console
+
+        self._console_status: SmartglassConsoleStatus = None
+        self._app_details: Optional[Product] = None
+
+    @property
+    def name(self):
+        """Return the device name."""
+        return self._console.name
+
+    @property
+    def unique_id(self):
+        """Console device ID."""
+        return self._console.id
+
+    @property
+    def state(self):
+        """State of the player."""
+        if self._console_status.playback_state in XBOX_STATE_MAP:
+            return XBOX_STATE_MAP[self._console_status.playback_state]
+        return XBOX_STATE_MAP[self._console_status.power_state]
+
+    @property
+    def supported_features(self):
+        """Flag media player features that are supported."""
+        active_support = SUPPORT_XBOX
+        if self.state not in [STATE_PLAYING, STATE_PAUSED]:
+            active_support &= ~SUPPORT_NEXT_TRACK & ~SUPPORT_PREVIOUS_TRACK
+        if not self._console_status.is_tv_configured:
+            active_support &= ~SUPPORT_VOLUME_MUTE & ~SUPPORT_VOLUME_STEP
+        return active_support
+
+    @property
+    def media_content_type(self):
+        """Media content type."""
+        if self._app_details and self._app_details.product_family == "Games":
+            return MEDIA_TYPE_GAME
+        return MEDIA_TYPE_APP
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        if self._app_details:
+            return self._app_details.localized_properties[0].short_title
+        return None
```

Comment:
If we make it a guard clause we'll save some line width.
```suggestion
        if not self._app_details:
        	return None
        return self._app_details.localized_properties[0].short_title
```